devtools/maven/src/main/java/io/quarkus/maven/RemoteDevMojo.java

```diff
@@ -1,132 +1,21 @@
-            return;
-        }
-
-        if (!sourceDir.isDirectory()) {
-            throw new MojoFailureException("The `src/main/java` directory is required, please create it.");
-        }
-
-        String resources = null;
-        for (Resource i : project.getBuild().getResources()) {
-            //todo: support multiple resources dirs for config hot deployment
-            resources = i.getDirectory();
-            break;
-        }
-
-        String classes = outputDirectory.getAbsolutePath();
-        String sources = sourceDir.getAbsolutePath();
-
-        //first lets look for some config, as it is not on the current class path
-        //and we need to load it to run the build process
-        if (resources != null) {
-            Path config = Paths.get(resources).resolve("application.properties");
-            if (Files.exists(config)) {
-                try {
-                    SmallRyeConfig built = ConfigUtils.configBuilder(false)
-                            .withSources(new PropertiesConfigSource(config.toUri().toURL())).build();
-                    QuarkusConfigFactory.setConfig(built);
-                    final ConfigProviderResolver cpr = ConfigProviderResolver.instance();
-                    final Config existing = cpr.getConfig();
-                    if (existing != built) {
-                        cpr.releaseConfig(existing);
-                        // subsequent calls will get the new config
-                    }
-                } catch (Exception e) {
-                    throw new RuntimeException(e);
-                }
-            }
-        }
-
-        Optional<String> url = ConfigProvider.getConfig().getOptionalValue("quarkus.live-reload.url", String.class);
-        Optional<String> password = ConfigProvider.getConfig().getOptionalValue("quarkus.live-reload.password",
-                String.class);
-        if (!url.isPresent()) {
-            throw new MojoFailureException("To use remote-dev you must specify quarkus.live-reload.url");
-        }
-        if (!password.isPresent()) {
-            throw new MojoFailureException("To use remote-dev you must specify quarkus.live-reload.password");
-        }
-        System.out.println(sources);
-        String remotePath = url.get();
-        if (remotePath.endsWith("/")) {
-            remotePath = remotePath.substring(0, remotePath.length() - 1);
-        }
-        AgentRunner runner = new AgentRunner(resources, sources, classes, remotePath + "/quarkus/live-reload",
-                password.get());
-
-        runner.run();
-        for (;;) {
-            try {
-                Thread.sleep(10000);
-            } catch (InterruptedException e) {
-                throw new RuntimeException(e);
-            }
-        }
+    protected void modifyDevModeContext(DevModeContext devModeContext) {
+        devModeContext.setMode(QuarkusBootstrap.Mode.PROD);
```

Comment:
Could you please clarify why you set PROD here?

Reply:
The first step is to synchronize the local output with the server. We set prod mode to generate a production application to sync. I will add a comment.